### PR TITLE
refactor: use orchestrator instead of docker logic in `run_local.go`

### PR DESCRIPTION
**Pull request by dannyrandall**
see the orchestrator in https://github.com/aws/copilot-cli/pull/5369

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/cli/run_local.go [hunk 1]
```diff
@@ -293,200 +280,124 @@ func (o *runLocalOpts) Execute() error {
...
-		},
+	envVars, err := o.getEnvVars(ctx, td)
+	if err != nil {
+		return orchestrator.Task{}, fmt.Errorf("get env vars: %w", err)
 	}
 
-	//channel to receive any error from the goroutine
-	errCh := make(chan error, 1)
+	task := orchestrator.Task{
+		Containers: make(map[string]orchestrator.ContainerDefinition),
```
Comment (Lou1415926) [inline internal/pkg/cli/run_local.go]: nitnitnit: 
```suggestion
		Containers: make(map[string]orchestrator.ContainerDefinition, len(td.ContainerDefinitions)),
```

File: internal/pkg/cli/run_local_test.go [hunk 2]
```diff
@@ -411,198 +398,71 @@ func TestRunLocalOpts_Execute(t *testing.T) {
...
+				errCh := make(chan error, 1)
+				m.orchestrator.StartFn = func() <-chan error {
+					errCh <- errors.New("some error")
+					return errCh
+				}
+				m.orchestrator.RunTaskFn = func(task orchestrator.Task) {
+					require.Equal(t, expectedTask, task)
+				}
+				m.orchestrator.StopFn = func() {
+					require.Len(t, errCh, 0)
```
Comment (Lou1415926) [inline internal/pkg/cli/run_local_test.go:428]: do you think it's necessary at all to test "how we are handling the error", which in our case, would mean "making sure we are really handling the error by printing it"?

File: internal/pkg/cli/run_local_test.go [hunk 3]
```diff
@@ -411,198 +398,71 @@ func TestRunLocalOpts_Execute(t *testing.T) {
...
-				m.dockerEngine.EXPECT().Stop(gomock.Any(), expectedRunPauseArgs.ContainerName).Return(nil)
-				m.dockerEngine.EXPECT().Rm(expectedRunPauseArgs.ContainerName).Return(nil)
+				}
+
+				count := 1
+				m.orchestrator.StopFn = func() {
+					switch count {
+					case 1:
+						errCh <- errors.New("error from stopping after sigint")
+					case 2:
```
Comment (Lou1415926) [inline internal/pkg/cli/run_local_test.go:457]: nice! I see we expect `runLocalOpts` to behaves like this:
1. Got SigInt, call `Stop()`.
2. Got err from `errCh`, call `Stop()` again.
3. Got err from `errCh`, call `Stop()` for the third time.

**Review by Lou1415926 — APPROVED** [review]
Looks great!

File: internal/pkg/cli/run_local_test.go (hunk 3 above)
Comment (dannyrandall) [reply internal/pkg/cli/run_local_test.go:457]: that's right! i'd also add:

4. exit after `errCh` is closed

Comment (iamhopaul123) [reply internal/pkg/cli/run_local_test.go:457]: :+1: this test case is awesome

**Review by iamhopaul123 — APPROVED** [review]

**Pull request merged**